libs/assigner_runner/src/utils.cpp

```diff
@@ -1,33 +1,43 @@
 #include "zkevm_framework/assigner_runner/utils.hpp"
 
-#include <evmc/evmc.h>
+#include <evmc.h>
 
 #include <cstdint>
 #include <fstream>
 #include <iomanip>
 #include <iostream>
 #include <sstream>
 
-evmc_call_kind evmc_msg_kind(const data_types::Transaction::Type& type) {
-    switch (type) {
-        case data_types::Transaction::Type::ContractCreation:
-            return EVMC_CREATE;
-        case data_types::Transaction::Type::MessageCall:
-            return EVMC_CALL;
-        default:
-            return EVMC_CALL;
-    };
+#include "zkevm_framework/core/types/message.hpp"
+
+evmc_call_kind evmc_msg_kind(const std::bitset<8>& type) {
+    if (type.test(std::size_t(core::types::MessageKind::Deploy))) {
+        return EVMC_CREATE;
+    } else if (type.test(std::size_t(core::types::MessageKind::Internal))) {
+        return EVMC_CALL;
+    } else {
+        return EVMC_CALL;
+    }
 }
 
-evmc_address to_evmc_address(const data_types::Address& v) {
+evmc_address to_evmc_address(const core::types::Address& v) {
     evmc_address res = {
-        v[0],  v[1],  v[2],  v[3],  v[4],  v[5],  v[6],  v[7],  v[8],  v[9],
-        v[10], v[11], v[12], v[13], v[14], v[15], v[16], v[17], v[18], v[19],
+        std::to_integer<uint8_t>(v[0]),  std::to_integer<uint8_t>(v[1]),
```

Comment:
Why couldn't you implement it via a loop?

Reply:
it's allow avoid initialization `evmc_address` as zero values array before filling.